pkg/file/loadsave/loadsave.go

```diff
@@ -21,16 +31,23 @@ type PutGetter interface {
 // package abstractions. use with caution since Loader will
 // load all of the subtrie of a given hash in memory.
 type loadSave struct {
-	storer    PutGetter
-	mode      storage.ModePut
-	encrypted bool
+	storer     PutGetter
+	pipelineFn func() pipeline.Interface
 }
 
-func New(storer PutGetter, mode storage.ModePut, enc bool) file.LoadSaver {
+// New returns a new read-write load-saver.
+func New(storer PutGetter, pipelineFn func() pipeline.Interface) file.LoadSaver {
```

Comment:
Seems like we already import `pipeline/builder here`. Would it be better to just instantiate the pipeline within this constructor?

That way we don't do a breaking change to the loadsave API.

Reply:
unfortunately this is not possible (at least not right now). every time you do a `save` you save an arbitrary length octet stream and so every save operation will result in a different root hash. a manifest can execute `save` more than once with the same loadsave instance. therefore, in order to reuse the same instance we need to implement a `reset` method on the pipeline. it would be nice to tackle this as a part of technical debt (as it is indeed a waste of resources to create a new pipeline every time. but then again - unless you're constantly uploading stuff it is a one-time-cost)